staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -899,9 +899,13 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = filterlatency.TrackStarted(handler, c.TracerProvider, "authentication")
 
 	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
+	handler = genericapifilters.WithWarningRecorder(handler)
```

Comment:
add a comment that this must be wrapped by the timeout handler to make the warning header additions threadsafe